guides/get-started-microservices-on-kubernetes/simple-microservice-database-part1.adoc

```diff
@@ -0,0 +1,397 @@
+                    <location>org.wildfly:wildfly-datasources-galleon-pack:{version-wildfly-datasources-galleon-pack}</location>
+                </feature-pack>
+            </feature-packs>
+            <layers>
+                <layer>cloud-server</layer>
+                <layer>postgresql-datasource</layer>
+            </layers>
+        </configuration>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>package</goal>
+                </goals>
+            </execution>
+        </executions>
+    </plugin>
+----
+
+=== Java Classes
+
+Add the following classes to the project:
+
+.org.wildfly.examples.test_table.model.TestTable   :
+[source,java]
+----
+package org.wildfly.examples.test_table.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+
+import java.util.Objects;
+
+import static jakarta.persistence.GenerationType.IDENTITY;
+
+@Entity
+public class TestTable {
+
+	@Id
+	@GeneratedValue(strategy = IDENTITY)
+	private Long id;
+
+	private String field1;
+
+	private String field2;
+
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+	}
+
+	public String getField1() {
+		return field1;
+	}
+
+	public void setField1(String field1) {
+		this.field1 = field1;
+	}
+
+	public String getField2() {
+		return field2;
+	}
+
+	public void setField2(String field2) {
+		this.field2 = field2;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		TestTable testTable = (TestTable) o;
+		return Objects.equals(id, testTable.id) && Objects.equals(field1, testTable.field1) && Objects.equals(field2, testTable.field2);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(id, field1, field2);
+	}
+}
+----
+
+.org.wildfly.examples.test_table.service.TestTableResource :
+[source,java]
+----
+package org.wildfly.examples.test_table.service;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.DefaultValue;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Response;
+import org.wildfly.examples.test_table.model.TestTable;
+
+import java.util.List;
+
+@RequestScoped
+@Path("/test-table")
+public class TestTableResource {
+
+	@PersistenceContext(unitName = "primary")
+	private EntityManager em;
+
+	public void createOrUpdateRow(TestTable row) {
+		if (!em.contains(row)) {
+			row = em.merge(row);
+		}
+		em.persist(row);
+	}
+
+	public List<TestTable> getAllRows() {
+		return em.createQuery("SELECT t FROM TestTable t", TestTable.class).getResultList();
+	}
+
+	@Transactional(Transactional.TxType.REQUIRED)
```

Comment:
Transactional.TxType.REQUIRED is the default one, there is no need to explicitly add it, remove it makes the code cleaner

Reply:
@yersan tnks, removed